pyvista/core/filters.py

```diff
@@ -2638,7 +2640,82 @@ def shrink(dataset, shrink_factor=1.0, progress_bar=False):
         if isinstance(dataset, _vtk.vtkPolyData):
             return output.extract_surface()
 
-    def reflect(dataset, normal, point=None, inplace=False):
+    def transform(dataset: _vtk.vtkDataSet,
+                  trans: Union[_vtk.vtkMatrix4x4, _vtk.vtkTransform, np.ndarray],
+                  transform_all_input_vectors=False, inplace=True):
+        """Transform this mesh with a 4x4 transform.
+
+        Parameters
+        ----------
+        trans : vtk.vtkMatrix4x4, vtk.vtkTransform, or np.ndarray
+            Accepts a vtk transformation object or a 4x4 transformation matrix.
+
+        transform_all_input_vectors: bool, optional
+            When ``True``, all input vectors are transformed. Otherwise, only the
+            points, normals and active vectors are transformed.
+        """
```

Comment:
Let's add an `Example` here, and I think we'll be good to go.  It really helps with the sphinx autodocs.